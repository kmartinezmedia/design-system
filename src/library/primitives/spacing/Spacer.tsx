import React, { FunctionComponent } from 'react';
import { View, ViewProps, StyleSheet } from 'react-native';

export const Spacer: FunctionComponent<ViewProps> = (props) => (
  <View {...props} style={styles.Spacer} />
);

const styles = StyleSheet.create({
  Spacer: {
    flexGrow: 1,
    flexShrink: 1,
    flexBasis: 'auto',
  },
});
