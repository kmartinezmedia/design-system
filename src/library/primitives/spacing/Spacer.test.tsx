import React from 'react';
import 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { Spacer } from './Spacer';

describe('Spacer', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<Spacer />);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
