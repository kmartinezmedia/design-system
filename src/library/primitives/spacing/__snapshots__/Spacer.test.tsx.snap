// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer default 1`] = `
<View
  style={
    Object {
      "flexBasis": "auto",
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
/>
`;
